File: frontend/projects/runtime/src/lib/events/task-executed-event.ts
import {BusEvent} from 'projects/event/src/lib/bus-event';
import {ExecutionEnvironment} from 'projects/runtime/src/lib/entities/execution-environment';

export class TaskExecutedEvent extends BusEvent {

  public static readonly CHANNEL = 'task-executed-event';

  constructor(public readonly taskId: string, public readonly context: ExecutionEnvironment) {
    super(TaskExecutedEvent.CHANNEL);
  }
}

File: backend/commons/runtime/runtime-event/src/main/java/com/kraken/runtime/event/TaskExecutedEvent.java
package com.kraken.runtime.event;

import com.kraken.runtime.context.entity.ExecutionContext;
import com.kraken.tools.event.bus.BusEvent;
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

@Value
@Builder
public class TaskExecutedEvent implements BusEvent {
  @NonNull ExecutionContext context;
}
